packages/agent/src/certificate.ts

```diff
@@ -21,20 +21,35 @@ interface Cert {
   delegation?: Delegation;
 }
 
-const enum NodeId {
-  Empty = 0,
-  Fork = 1,
-  Labeled = 2,
-  Leaf = 3,
-  Pruned = 4,
-}
+const NodeId = {
+  Empty: 0,
+  Fork: 1,
+  Labeled: 2,
+  Leaf: 3,
+  Pruned: 4,
+};
+
+export type NodeIdType = typeof NodeId[keyof typeof NodeId];
+
+export { NodeId };
 
 export type HashTree =
-  | [NodeId.Empty]
-  | [NodeId.Fork, HashTree, HashTree]
-  | [NodeId.Labeled, ArrayBuffer, HashTree]
-  | [NodeId.Leaf, ArrayBuffer]
-  | [NodeId.Pruned, ArrayBuffer];
+  | [typeof NodeId.Empty]
```

Comment:
how did this not produce a type error before?

Reply:
The JS agent's decoding logic simply didn't cover this case. The `/subnet` path never decoded correctly. 

At some point, the entire certificate ought to be re-written, modeled after the rust library for comprehensiveness